test/extensions/filters/http/common/compressor/compressor_filter_test.cc

```diff
@@ -233,6 +237,98 @@ TEST_F(CompressorFilterTest, CompressRequest) {
   doResponseNoCompression(headers);
 }
 
+TEST_F(CompressorFilterTest, CompressRequestNoContentLength) {
+  setUpFilter(R"EOF(
+{
+  "request_direction_config": {},
+  "compressor_library": {
+     "name": "test",
+     "typed_config": {
+       "@type": "type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip"
+     }
+  }
+}
+)EOF");
+  doRequestNoCompression({{":method", "post"}});
+  Http::TestResponseHeaderMapImpl headers{{":method", "post"}};
```

Comment:
Nit: It is a bit weird to use the `:method` header in response headers, I guess this is was copied from other tests. If it is not too much trouble can it be changes to `:status` please? Just to reduce confusion a bit.

Reply:
sure. as i will bake another diff w/ removing `isResponseWithContent` anyway.